realm/realm-library/src/main/java/io/realm/MutableRealmObjectSchema.java

```diff
@@ -87,6 +87,10 @@ public RealmObjectSchema addField(String fieldName, Class<?> fieldType, FieldAtt
         if (metadata == null) {
             if (SUPPORTED_LINKED_FIELDS.containsKey(fieldType)) {
                 throw new IllegalArgumentException("Use addRealmObjectField() instead to add fields that link to other RealmObjects: " + fieldName);
+            } else if (RealmModel.class.isAssignableFrom(fieldType)) {
```

Comment:
The issue also mention that we should document this in the Javadoc

Reply:
it has been documented already:
```
* To add fields that reference other RealmObjects or RealmLists use
     * {@link #addRealmObjectField(String, RealmObjectSchema)} or {@link #addRealmListField(String, RealmObjectSchema)}
     * instead.
```

how ever, there is a bug in the java doc, will fix it.